packages/location-management/src/components/LocationUnitView/index.tsx

```diff
@@ -130,52 +130,48 @@ export async function getHierarchy(location: LocationUnit[], opensrpBaseURL: str
 }
 
 export const LocationUnitView: React.FC<Props> = (props: Props) => {
-  const accessToken = useSelector((state) => getAccessToken(state) as string);
-  const treeData = (useSelector((state) =>
-    getAllHierarchiesArray(state)
-  ) as unknown) as ParsedHierarchyNode[];
   const dispatch = useDispatch();
+  const treeData = useSelector((state) => getAllHierarchiesArray(state));
+  const locationUnits = useSelector((state) => getLocationUnitsArray(state));
   const [tableData, setTableData] = useState<TableData[]>([]);
   const [detail, setDetail] = useState<LocationDetailData | 'loading' | null>(null);
-  const [currentParentChildren, setCurrentParentChildren] = useState<ParsedHierarchyNode[]>([]);
   const [currentClicked, setCurrentClicked] = useState<ParsedHierarchyNode | null>(null);
   const { opensrpBaseURL } = props;
 
   useEffect(() => {
-    if (!treeData.length) {
+    if (!locationUnits.length) {
       getBaseTreeNode(opensrpBaseURL)
-        .then((response) => {
-          dispatch(fetchLocationUnits(response));
-          getHierarchy(response, opensrpBaseURL)
-            .then((hierarchy) => {
-              hierarchy.forEach((hier) => {
-                const processed = generateJurisdictionTree(hier);
-                dispatch(fetchAllHierarchies(processed.model));
-              });
-            })
-            .catch(() => sendErrorNotification(ERROR_OCCURED));
+        .then((response) => dispatch(fetchLocationUnits(response)))
+        .catch(() => sendErrorNotification(ERROR_OCCURED));
+    }
+  }, [locationUnits.length, dispatch, opensrpBaseURL]);
+
+  useEffect(() => {
+    if (!treeData.length && locationUnits.length) {
+      getHierarchy(locationUnits, opensrpBaseURL)
+        .then((hierarchy) => {
+          const allhierarchy = hierarchy.map((hier) => generateJurisdictionTree(hier).model);
+          dispatch(fetchAllHierarchies(allhierarchy));
         })
         .catch(() => sendErrorNotification(ERROR_OCCURED));
     }
-  }, [treeData, accessToken, dispatch, opensrpBaseURL]);
+    // to avoid extra rerenders
+    // eslint-disable-next-line react-hooks/exhaustive-deps
+  }, [locationUnits.length, treeData.length, dispatch, opensrpBaseURL]);
 
   useEffect(() => {
-    const data = parseTableData(currentParentChildren.length ? currentParentChildren : treeData);
-    setTableData(data);
-  }, [treeData, currentParentChildren]);
-
-  if (!tableData.length || !treeData.length)
-    return (
-      <Spin
-        style={{
-          display: 'flex',
-          alignItems: 'center',
-          justifyContent: 'center',
-          minHeight: '85vh',
-        }}
-        size={'large'}
-      />
-    );
+    if (treeData.length) {
+      if (currentClicked && currentClicked.children) {
```

Comment:
Looks like we could use some comments here to explain this block of useEffect

Reply:
Just Added description to this block